docs/notes/2.23.x.md

```diff
@@ -18,6 +18,9 @@ We offer [formal sponsorship tiers for companies](https://www.pantsbuild.org/spo
 
 ### Backends
 
+#### Python
+
+[The `pants.backend.experimental.python.typecheck.pyright` backend](https://www.pantsbuild.org/2.23/reference/subsystems/pyright) now uses version 1.1.365 by default.
```

Comment:
This does not belong in this MR?

I'm also expecting the version of defaults (in a Javacript section) to be listed here so that users gets a chance to react.

Reply:
Sorry about that. I branched this off of another MR that I was working on at the time (#21007) so this got included by accident. I'll remove the commits as that MR has been merged now.

Reply:
And this and the other unrelated stuff should be gone now.